v2/yarpcfx/yarpcfx_test.go

```diff
@@ -167,3 +168,43 @@ func TestNewRouter(t *testing.T) {
 	require.NoError(t, err)
 	assert.Len(t, res.Router.Procedures(), 3)
 }
+
+func TestNewRouterWithInboundMiddleware(t *testing.T) {
+	var callOrder []string
+
+	unaryMiddleware := []yarpc.UnaryInboundTransportMiddleware{
+		yarpc.UnaryInboundTransportMiddlewareFunc(
+			func(ctx context.Context, _ *yarpc.Request, _ *yarpc.Buffer, h yarpc.UnaryTransportHandler) (*yarpc.Response, *yarpc.Buffer, error) {
+				callOrder = append(callOrder, "middleware")
+				return h.Handle(ctx, nil, nil)
+			}),
+	}
+
+	handler := yarpc.UnaryTransportHandlerFunc(
+		func(context.Context, *yarpc.Request, *yarpc.Buffer) (*yarpc.Response, *yarpc.Buffer, error) {
+			callOrder = append(callOrder, "handler")
+			return nil, nil, nil
+		})
+
+	procedures := []yarpc.TransportProcedure{
+		{
+			Name:        "foo",
+			HandlerSpec: yarpc.NewUnaryTransportHandlerSpec(handler),
+		},
+	}
+
+	result, err := NewRouter(RouterParams{
+		Procedures:               procedures,
+		UnaryTransportMiddleware: unaryMiddleware,
```

Comment:
This is happening at the Fx layer, specifically in `yarpfx`, just before creating the `MapRouter`.
I'd rather keep `yarpcfx.NewRouter` as is because `yarpcfx.New` seems a bit ambiguous.

Maybe I'm misunderstanding you?

Reply:
@kriskowal Might be suggesting that the `[]yarpc.TransportProcedure` should already be well-formed / have their middleware applied before they are provided into the `yarpc.Router` constructor.